tests/examples/19_is_defined.yml

```diff
@@ -15,7 +15,8 @@
       condition: event.msg is defined
       action:
         debug:
-
-
-
-
+    - name: r3
+      condition: event.payload is defined
```

Comment:
This rulebook uses `range` as the source. Neither the `range` source nor the `set_fact` action ever generates `payload`. Although you test the rule in `test_examples` by manually creating a `payload`, will it be better to move the rule to another rulebook or add another `set_fact` to create a `payload`?

Reply:
@bze We have a lot of test cases where the event is injected into the Queue directly without having the source plugins to do that. We don't have a source plugin that doesn't insert anything and allows others to insert the data. The data insertion is controlled by the tests.